src/Cli/dotnet/Installer/Windows/InstallerBase.cs

```diff
@@ -8,201 +8,200 @@
+    protected bool VerifySignatures
+    {
+        get;
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="InstallerBase"/> instance using the specified elevation context and logger.
+    /// </summary>
+    /// <param name="elevationContext"></param>
+    /// <param name="logger"></param>
+    /// <param name="verifySignatures">Determines whether MSI signatures should be verified</param>
+    protected InstallerBase(InstallElevationContextBase elevationContext, ISetupLogger logger, bool verifySignatures)
+    {
+        ElevationContext = elevationContext;
+        Log = logger;
+        VerifySignatures = verifySignatures;
+    }
+
+    /// <summary>
+    /// Starts an elevated process to perform privileged operations.
+    /// </summary>
+    protected void Elevate()
+    {
+        ElevationContext.Elevate();
+    }
+
+    /// <summary>
+    /// Checks the specified error code to determine whether it indicates a success result. If not, additional extended information
+    /// is retrieved before throwing a <see cref="WorkloadException"/>.
+    /// 
+    /// The <see cref="Restart"/> property will be set to <see langword="true" /> if the error is either <see cref="Error.SUCCESS_REBOOT_INITIATED"/>
+    /// or <see cref="Error.SUCCESS_REBOOT_REQUIRED"/>.
+    /// </summary>
+    /// <param name="error">The error code to check.</param>
+    /// <param name="message">The message to include the exception.</param>
+    /// <exception cref="WorkloadException" />
+    protected void ExitOnError(uint error, string message)
+    {
+        if (!Error.Success(error))
         {
-            if (response.HResult < 0)
-            {
-                throw new WorkloadException(string.IsNullOrWhiteSpace(message) ? response.Message : $"{message} {response.Message}", response.HResult);
-            }
+            throw new WorkloadException($"{message} Error: 0x{error:x8}, {Marshal.GetPInvokeErrorMessage((int)error)}");
         }
 
-        /// <summary>
-        /// Logs a message if the specified error code does not indicate a success result. The <see cref="Restart"/>
-        /// property will be set to <see langword="true" /> if the error is either <see cref="Error.SUCCESS_REBOOT_INITIATED"/>
-        /// or <see cref="Error.SUCCESS_REBOOT_REQUIRED"/>.
-        /// 
-        /// No exception is thrown by this method. See <see cref="ExitOnError(uint, string)"/> for more detail.
-        /// </summary>
-        /// <param name="error">The error code to log.</param>
-        /// <param name="message">The message to log.</param>
-        protected void LogError(uint error, string message)
-        {
-            if (!Error.Success(error))
-            {
-                Log?.LogMessage($"{message} Error: 0x{error:x8}.");
-            }
+        // Once set to true, we retain restart information for the duration of the underlying command.
+        Restart |= error == Error.SUCCESS_REBOOT_INITIATED || error == Error.SUCCESS_REBOOT_REQUIRED;
```

Comment:
It is gonna keep doing this, @Forgind . I'm just gonna let it do it.

Note: It removed parentheses. I cannot seem to highlight the before/after correctly because of ignoring whitespace when viewing the PR.